artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/FileLockNodeManager.java

```diff
@@ -338,5 +345,104 @@ protected FileLock lock(final long lockPosition) throws Exception {
       while (lock == null);
       return lock;
    }
+   
+   private void startLockMonitoring()	{
+	   logger.debug("Starting the lock monitor");
+	   Thread monitorThread = new Thread(new MonitorLock());
```

Comment:
We have thread executors. It is important it must use these. Also this solution uses up a thread permanently which goes agaisnt some of the thread modelling. Look maybe to use schedule thread executor